### PR TITLE
Use the marketplace builder in dev node

### PR DESCRIPTION
**Pull request by ImJeremyHe**
Currently in this [dae071b](https://github.com/EspressoSystems/espresso-sequencer/pull/1983/commits/dae071bf41dccd58248c5019e5315b876baf6c81) we are referring to a branch not a tag from the `marketplace-builder-core`. Should replace this with the tag after [this PR](https://github.com/EspressoSystems/marketplace-builder-core/pull/123) has been merged.

### This PR:
- Specify the sequencer version to use the marketplace builder in dev node
- Fix the builder listener so that the builder API is accessible when running in a docker container

### CODE REVIEW

File: sequencer/src/bin/espresso-dev-node.rs
Comment (imabdulbasit) [inline sequencer/src/bin/espresso-dev-node.rs:193]: Do we want to run the dev node with the marketplace version or have an argument to run v1/fee/marketplace?

Comment (ImJeremyHe) [reply sequencer/src/bin/espresso-dev-node.rs]: I am not sure. dev-node is for the developers who want to integrate their rollups with espresso. If you think they have to use this function, then it will be necessary for us to add this function.

File: Cargo.toml
Comment (sveitser) [inline Cargo.toml:61]: This should be updated before merge so we don't use a branch.

Comment (imabdulbasit) [reply Cargo.toml]: done in 2f636f46f22e5c1a80d633326ec8456f11fb4281

File: sequencer/src/bin/espresso-dev-node.rs
Comment (tbro) [reply sequencer/src/bin/espresso-dev-node.rs]: I think we should support any version. Marketplace will not be live for some time.

Comment (imabdulbasit) [reply sequencer/src/bin/espresso-dev-node.rs]: This was discussed in standup and I think we only need marketplace version here cc @sveitser

Comment (tbro) [reply sequencer/src/bin/espresso-dev-node.rs]: Ok. I think it limits the usefulness of devnode, but I also don't remember the conversation. I go ahead a approve and if we need it we can add versions later.

**Review by tbro — APPROVED** [review]

File: sequencer/src/bin/espresso-dev-node.rs
Comment (imabdulbasit) [reply sequencer/src/bin/espresso-dev-node.rs]: I do agree tho that we should have support for pre marketplace version so I'll take a look at it next week. I think this blocks some other work so should be merged

Comment (imabdulbasit) [reply sequencer/src/bin/espresso-dev-node.rs]: https://github.com/EspressoSystems/espresso-sequencer/issues/2033

**Pull request merged**
